handlers/singular.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+const parse = require('../lib/parse');
+const stringifyError = require('../stringify-error');
+
+module.exports = function createSingularHandler(bundler, options) {
+  options = options || {};
+
+  return function singularHandler(req, res) {
+    const serve = serveBundle(res);
+    const fail = fiveHundred(res);
+    const ack = acknowledgePurge(res);
+
+    let input;
+    try {
+      input = parse(req.params.slug, options);
+    }
+    catch (err) {
+      return fail(err);
+    }
+
+    function done(p) {
+      return p.catch((err) => setImmediate(() => { throw err; }));
```

Comment:
this will end up returning a resolved promise. you may want `p.catch(…); return p;`?